engine/components-api/src/com/cloud/agent/AgentManager.java

```diff
@@ -141,7 +142,7 @@
 
     public void pullAgentOutMaintenance(long hostId);
 
-    boolean reconnect(long hostId);
+    void reconnect(long hostId) throws CloudRuntimeException, AgentUnavailableException;
```

Comment:
@rafaelweingartner 
It is just a heads up to tell them hey this method can also throw a runtimeException and i think it needs to be handled by the caller. In way it acts like a lable to do further refactoring.  If you see the implementation of reconnect in clusteredAgentMangerImpl, it throws runtime exceptions which tell the reason why it failed to reconnect, but since it is a runtimeException no one care to check it and hence the reason for failure is not propagated to the caller, and the API will just say failed to reconnect without giving any reason for the failure.